src/Serialization.js

```diff
@@ -89,11 +96,14 @@ exports.newVkeyFromPublicKey = public_key => () => lib.Vkey.new(public_key);
 exports._publicKeyFromBech32 = maybe => bech32 => {
   // this is needed because try/catch overuse breaks runtime badly
   // https://github.com/Plutonomicon/cardano-transaction-lib/issues/875
-  // prefixes defined in CIP-5: https://github.com/cardano-foundation/CIPs/tree/master/CIP-0005 
-  if (/^(acct|acct_shared|addr|addr_shared|policy|pool|root|root_shared|stake|stake_shared)_x?vk1[0-9a-z]+$/.test(bech32)) {
-      return maybe.just(lib.BigInt.from_str(bech32));
-  } else {
-      return maybe.nothing;
+  try {
+    if (/^ed25519_pk1[0-9a-z]+$/.test(bech32)) {
```

Comment:
Why `pk1` and not `pk`? I'm not sure about this, have you found the code it originates from?

Reply:
@klntsky 1 is the separator for bech32, so this is fine.

@pb99u069 The payload (hex part) must be 32 bytes long
https://github.com/Emurgo/cardano-serialization-lib/blob/04c4e7a0ee1e65253a47543be303740902820156/rust/src/chain_crypto/algorithms/ed25519.rs#L55
https://docs.rs/ed25519-dalek/0.8.0/ed25519_dalek/constant.PUBLIC_KEY_LENGTH.html

So we can also check if the string is `32/4+"ed25519_pk1".length` long

Reply:
@klntsky 1 is the separator for bech32, so this is fine.

@pb99u069 The payload (hex part) must be 32 bytes long
https://github.com/Emurgo/cardano-serialization-lib/blob/04c4e7a0ee1e65253a47543be303740902820156/rust/src/chain_crypto/algorithms/ed25519.rs#L55
https://docs.rs/ed25519-dalek/0.8.0/ed25519_dalek/constant.PUBLIC_KEY_LENGTH.html

So we can also check if the string is `32*8/4+"ed25519_pk1".length` long

Reply:
The maths isn't that simple (nor is it hex)